### PR TITLE
fix: load compliance report from file system

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>

## Description
Load compliance report from file system

## Related issues
- Related #2919

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).

### CODE REVIEW

File: pkg/compliance/spec/compliance.go [hunk 1]
```diff
@@ -104,7 +105,7 @@ func GetComplianceSpec(specNameOrPath string) (string, error) {
 	if strings.HasPrefix(specNameOrPath, "@") {
 		buf, err := os.ReadFile(strings.TrimPrefix(specNameOrPath, "@"))
 		if err != nil {
-			return "", fmt.Errorf("error retrieving complaince spec from path: %w", err)
+			return "", fmt.Errorf("error retrieving compliancelaince spec from path: %w", err)
```
Comment (josedonizetti) [inline pkg/compliance/spec/compliance.go:108]: typo? `compliance`?

**Review by josedonizetti — APPROVED** [review]
Looks good, only one typo to fix.

File: pkg/flag/report_flags.go [hunk 2]
```diff
@@ -217,7 +217,7 @@ func (f *ReportFlagGroup) ToOptions(out io.Writer) (ReportOptions, error) {
 
 func parseComplianceTypes(compliance interface{}) (string, error) {
```
Comment (knqyf263) [inline pkg/flag/report_flags.go:218]: I think `compliance` can be `string` as we already call `getString`.

File: pkg/compliance/spec/compliance.go [hunk 3]
```diff
@@ -96,3 +99,16 @@ func securityCheckByCheckID(checkID string) types.SecurityCheck {
...
 }
+
+// GetComlianceSpec accepct compliance flag name/path and return builtin or file system loaded spec
+func GetComplianceSpec(specNameOrPath string) (string, error) {
+	if strings.HasPrefix(specNameOrPath, "@") {
+		buf, err := os.ReadFile(strings.TrimPrefix(specNameOrPath, "@"))
+		if err != nil {
+			return "", fmt.Errorf("error retrieving compliance spec from path: %w", err)
+		}
+		return string(buf), nil
```
Comment (knqyf263) [inline pkg/compliance/spec/compliance.go:110]: What if returning `[]byte` as is? This file content is converted to `string` and then will be converted to `[]byte` just after this function. Seems like it is better to convert the return value of `GetSpecByName` to `[]byte`.

File: docs/docs/kubernetes/cli/compliance.md [hunk 4]
```diff
@@ -66,3 +66,52 @@ $ trivy k8s cluster --compliance=nsa --report summary --format json
...
+
+
+```yaml
+---
+spec:
+  id: "0001" # report unique identifier
+  title: nsa # report title 
+  description: National Security Agency - Kubernetes Hardening Guidance # description of the report
+  relatedResources :
+    - https://www.nsa.gov/Press-Room/News-Highlights/Article/Article/2716980/nsa-cisa-release-kubernetes-hardening-guidance/ # refreance is related to public or internal spec
```
Comment (knqyf263) [inline docs/docs/kubernetes/cli/compliance.md:92]:
```suggestion
    - https://www.nsa.gov/Press-Room/News-Highlights/Article/Article/2716980/nsa-cisa-release-kubernetes-hardening-guidance/ # reference is related to public or internal spec
```

File: docs/docs/kubernetes/cli/compliance.md [hunk 5]
```diff
@@ -66,3 +66,52 @@ $ trivy k8s cluster --compliance=nsa --report summary --format json
...
+      description: 'Check that container root file system is immutable'
+      id: '1.1'
+      checks:
+        - id: AVD-KSV-0014
+      severity: 'LOW'
+```
+
+## Custom report CLI Commands
+
+To generate the custom report an custom spec file path should be passed to the trivy command with `@` prefix as follow:
```
Comment (knqyf263) [inline docs/docs/kubernetes/cli/compliance.md:111]: nit
```suggestion
To generate the custom report, an custom spec file path should be passed to the `--compliance` flag with `@` prefix as follow:
```

**Comment by chen-keinan** [comment]
@knqyf263 comments has been addressed

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**